wpilibc/src/main/native/include/frc/Alert.h

```diff
@@ -0,0 +1,123 @@
+ * periodically.
+ *
+ * <pre>
+ * class Robot {
+ *   frc::Alert alert{"Something went wrong", frc::Alert::AlertType::kWarning};
+ * }
+ *
+ * Robot::periodic() {
+ *   alert.Set(...);
+ * }
+ * </pre>
+ */
+class Alert {
+ public:
+  /**
+   * Represents an alert's level of urgency.
+   */
+  enum class AlertType {
+    /**
+     * High priority alert - displayed first on the dashboard with a red "X"
+     * symbol. Use this type for problems which will seriously affect the
+     * robot's functionality and thus require immediate attention.
+     */
+    kError,
+
+    /**
+     * Medium priority alert - displayed second on the dashboard with a yellow
+     * "!" symbol. Use this type for problems which could affect the robot's
+     * functionality but do not necessarily require immediate attention.
+     */
+    kWarning,
+
+    /**
+     * Low priority alert - displayed last on the dashboard with a green "i"
+     * symbol. Use this type for problems which are unlikely to affect the
+     * robot's functionality, or any other alerts which do not fall under the
+     * other categories.
+     */
+    kInfo
+  };
+
+  /**
+   * Creates a new alert in the default group - "Alerts". If this is the first
+   * to be instantiated, the appropriate entries will be added to NetworkTables.
+   *
+   * @param text Text to be displayed when the alert is active.
+   * @param type Alert urgency level.
+   */
+  Alert(std::string_view text, AlertType type);
+
+  /**
+   * Creates a new alert. If this is the first to be instantiated in its group,
+   * the appropriate entries will be added to NetworkTables.
+   *
+   * @param group Group identifier, used as the entry name in NetworkTables.
+   * @param text Text to be displayed when the alert is active.
+   * @param type Alert urgency level.
+   */
+  Alert(std::string_view group, std::string_view text, AlertType type);
+
+  /**
+   * Sets whether the alert should currently be displayed. When activated, the
+   * alert text will also be sent to the console. This method can be safely
+   * called periodically.
+   *
+   * @param active Whether to display the alert.
+   */
+  void Set(bool active);
+
+  /**
+   * Updates current alert text. Use this method to dynamically change the
+   * displayed alert, such as including more details about the detected problem.
+   *
+   * @param text Text to be displayed when the alert is active.
+   */
+  void SetText(std::string_view text);
+
+ private:
+  AlertType m_type;
+  bool m_active = false;
+  units::second_t m_activeStartTime;
+  std::string_view m_text;
+
+  class SendableAlerts : public nt::NTSendable,
+                         public wpi::SendableHelper<SendableAlerts> {
+   public:
+    std::vector<std::shared_ptr<Alert>> m_alerts;
+    void InitSendable(nt::NTSendableBuilder& builder) override;
+
+   private:
+    std::vector<std::string> GetStrings(AlertType type);
+  };
+
+  static std::map<std::string_view, SendableAlerts> groups;
```

Comment:
Is there a reasonable upper bound on number of elements that would allow using [`wpi::SmallDenseMap`](https://github.wpilib.org/allwpilib/docs/release/cpp/classwpi_1_1_small_dense_map.html)?

Reply:
It's almost certainly unsafe to use a map with a string_view key.  Use `wpi::StringMap` instead.